src/annotator/integrations/test/vitalsource-test.js

```diff
@@ -0,0 +1,215 @@
+import { delay } from '../../../test-util/wait';
+import {
+  VitalSourceContainerIntegration,
+  VitalSourceContentIntegration,
+  vitalSourceFrameRole,
+  $imports,
+} from '../vitalsource';
+
+class FakeVitalSourceViewer {
+  constructor() {
+    this.bookElement = document.createElement('mosaic-book');
+    this.bookElement.attachShadow({ mode: 'open' });
+
+    this.contentFrame = document.createElement('iframe');
+    this.bookElement.shadowRoot.append(this.contentFrame);
+
+    document.body.append(this.bookElement);
+  }
+
+  destroy() {
+    this.bookElement.remove();
+  }
+
+  /** Simulate navigation to a different chapter of the book. */
+  loadNextChapter() {
+    this.contentFrame.remove();
+
+    // VS handles navigations by removing the frame and creating a new one,
+    // rather than navigating the existing frame.
+    this.contentFrame = document.createElement('iframe');
+    this.bookElement.shadowRoot.append(this.contentFrame);
+  }
+}
+
+describe('annotator/integrations/vitalsource', () => {
+  let fakeViewer;
+  let FakeHTMLIntegration;
+  let fakeHTMLIntegration;
+
+  beforeEach(() => {
+    fakeViewer = new FakeVitalSourceViewer();
+
+    fakeHTMLIntegration = {
+      anchor: sinon.stub(),
+      contentContainer: sinon.stub(),
+      describe: sinon.stub(),
+      destroy: sinon.stub(),
+      scrollToAnchor: sinon.stub(),
+    };
+
+    FakeHTMLIntegration = sinon.stub().returns(fakeHTMLIntegration);
+
+    $imports.$mock({
+      './html': { HTMLIntegration: FakeHTMLIntegration },
+    });
+  });
+
+  afterEach(() => {
+    fakeViewer.destroy();
+    $imports.$restore();
+  });
+
+  describe('vitalSourceFrameRole', () => {
+    it('returns "container" if book container element is found', () => {
+      assert.equal(vitalSourceFrameRole(), 'container');
+    });
+
+    it('returns "content" if the book container element is found in the parent document', () => {
+      assert.equal(
+        vitalSourceFrameRole(fakeViewer.contentFrame.contentWindow),
+        'content'
+      );
+    });
+
+    it('returns `null` if the book container element is not found', () => {
+      fakeViewer.destroy();
+      assert.isNull(vitalSourceFrameRole());
+    });
+  });
+
+  describe('VitalSourceContainerIntegration', () => {
+    let fakeGuest;
+    let integration;
+
+    beforeEach(() => {
+      fakeGuest = {
+        injectClient: sinon.stub(),
+      };
```

Comment:
This highlights the slight confusion between `Guest` (the class) and `Annotator` (the interface).